From dda485a3d788c47bb6849218f695db7b51822872 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Tue, 31 Oct 2017 14:51:51 +0200
Subject: [PATCH] btrfs-progs: convert: add missing types header

Build with musl libc needs the sys/types.h header for the dev_t type,
since this header is not included indirectly. This fixes the following
build failure:

In file included from convert/source-fs.c:23:0:
./convert/source-fs.h:112:1: error: unknown type name ‘dev_t’
 dev_t decode_dev(u32 dev);
 ^~~~~
convert/source-fs.c:31:1: error: unknown type name ‘dev_t’
 dev_t decode_dev(u32 dev)
 ^~~~~

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: https://marc.info/?l=linux-btrfs&m=150945479718731&w=2

 convert/source-fs.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/convert/source-fs.h b/convert/source-fs.h
index 23f33567b42d..4e5babefe6a2 100644
--- a/convert/source-fs.h
+++ b/convert/source-fs.h
@@ -19,6 +19,7 @@
 
 #include "kerncompat.h"
 #include <linux/kdev_t.h>
+#include <sys/types.h>
 #include <pthread.h>
 
 #define CONV_IMAGE_SUBVOL_OBJECTID BTRFS_FIRST_FREE_OBJECTID
-- 
2.14.2

